doc/stdenv/platform-notes.chapter.md

```diff
@@ -2,242 +2,317 @@
+  A version of `xcrun` is part of the `stdenv` on Darwin.
+  If your package invokes `xcrun` via an absolute path (such as `/usr/bin/xcrun`), you will need to patch the build scripts to use `xcrun` instead.
+
+To reiterate: you usually don’t have to worry about this stuff.
+Start with writing your derivation as if everything is already set up for you (because in most cases it already is).
+If you run into issues or failures, continue reading below for how to deal with the most common issues you may encounter.
+
+### Darwin Issue Troubleshooting {#sec-darwin-troubleshooting}
+
+#### `xcrun` cannot find a binary {#sec-darwin-troubleshooting-xcrun}
+
+`xcrun` searches `PATH` and the SDK’s toolchain for binaries to run.
+If it cannot find a required binary, it will fail. When that happens, add the package for that binary to your derivation’s `nativeBuildInputs` (or `nativeCheckInputs` if the failure is happening when running tests).
+
+```nix
+stdenv.mkDerivation {
+  name = "libfoo-1.2.3";
+  # ...
+  nativeBuildInputs = [ bison ];
+  buildCommand = ''
+    xcrun bison foo.y # produces foo.tab.c
     # ...
-    buildPhase = ''
-      $CC -o hello hello.c
-    '';
-  }
-  ```
+  '';
+}
+```
+
+#### Package requires `xcodebuild` {#sec-darwin-troubleshooting-xcodebuild}
+
+The xcbuild package provides an `xcodebuild` command for packages that really depend on Xcode.
+This replacement is not 100% compatible and may run into some issues, but it is able to build many packages.
+To use `xcodebuild`, add `xcbuildHook` to your package’s `nativeBuildInputs`.
+It will provide a `buildPhase` for your derivation.
+You can use `xcbuildFlags` to specify flags to `xcodebuild` such as the required schema.
+If a schema has spaces in its name, you must set `__structuredAttrs` to `true`.
+See MoltenVK for an example of setting up xcbuild.
+
+```nix
+stdenv.mkDerivation {
+  name = "libfoo-1.2.3";
+  xcbuildFlags = [
+    "-configuration"
+    "Release"
+    "-project"
+    "libfoo-project.xcodeproj"
+    "-scheme"
+    "libfoo Package (macOS only)"
+  ];
+  __structuredAttrs = true;
+}
+```
+
+##### Fixing absolute paths to `xcodebuild`, `xcrun`, and `PListBuddy` {#sec-darwin-troubleshooting-xcodebuild-absolute-paths}
+
+Many build systems hardcode the absolute paths to `xcodebuild`, `xcrun`, and `PListBuddy` as `/usr/bin/xcodebuild`, `/usr/bin/xcrun`, and `/usr/bin/PListBuddy` respectively.
+These paths will need to be replaced with relative paths and the xcbuild package if `xcodebuild` or `PListBuddy` are used.
+
+```nix
+stdenv.mkDerivation {
+  name = "libfoo-1.2.3";
+  postPatch = ''
+    subsituteInPlace Makefile \
+      --replace-fail '/usr/bin/xcodebuild' 'xcodebuild' \
+      --replace-fail '/usr/bin/xcrun' 'xcrun' \
+      --replace-fail '/usr/bin/PListBuddy' 'PListBuddy'
+  '';
+}
+```
+
+#### Package requires a non-default SDK or fails to build due to missing frameworks or symbols {#sec-darwin-troubleshooting-using-sdks}
```

Comment:
I’d be partial to moving this first, since I think it’s more common than `xcrun` issues or packages that use Xcode build systems, especially since it often affects non‐Darwin‐specific software, but will leave this up to your judgement.